tests/e2e/p/staking_rewards.go

```diff
@@ -68,68 +75,50 @@ var _ = ginkgo.Describe("[Staking Rewards]", func() {
 		betaNodeID, betaPOP, err := betaInfoClient.GetNodeID(tc.DefaultContext())
 		require.NoError(err)
 
-		tc.By("generating reward keys")
-
-		alphaValidationRewardKey, err := secp256k1.NewPrivateKey()
-		require.NoError(err)
-		alphaDelegationRewardKey, err := secp256k1.NewPrivateKey()
-		require.NoError(err)
-
-		betaValidationRewardKey, err := secp256k1.NewPrivateKey()
-		require.NoError(err)
-		betaDelegationRewardKey, err := secp256k1.NewPrivateKey()
-		require.NoError(err)
-
-		gammaDelegationRewardKey, err := secp256k1.NewPrivateKey()
-		require.NoError(err)
-
-		deltaDelegationRewardKey, err := secp256k1.NewPrivateKey()
-		require.NoError(err)
+		tc.By("creating keychain and P-Chain wallet")
+		var (
+			alphaValidationRewardKey = e2e.NewKey(tc)
+			alphaDelegationRewardKey = e2e.NewKey(tc)
+			betaValidationRewardKey  = e2e.NewKey(tc)
+			betaDelegationRewardKey  = e2e.NewKey(tc)
+			gammaDelegationRewardKey = e2e.NewKey(tc)
+			deltaDelegationRewardKey = e2e.NewKey(tc)
+
+			rewardKeys = []*secp256k1.PrivateKey{
+				alphaValidationRewardKey,
+				alphaDelegationRewardKey,
+				betaValidationRewardKey,
+				betaDelegationRewardKey,
+				gammaDelegationRewardKey,
+				deltaDelegationRewardKey,
+			}
 
-		rewardKeys := []*secp256k1.PrivateKey{
-			alphaValidationRewardKey,
-			alphaDelegationRewardKey,
-			betaValidationRewardKey,
-			betaDelegationRewardKey,
-			gammaDelegationRewardKey,
-			deltaDelegationRewardKey,
-		}
+			keychain = env.NewKeychain(1)
+			nodeURI  = tmpnet.NodeURI{
+				NodeID: alphaNodeID,
+				URI:    alphaNode.URI,
+			}
+			baseWallet = e2e.NewWallet(tc, keychain, nodeURI)
```

Comment:
This is the actual change. The keychain being passed into the wallet here only has the single funded key.

This doesn't change any behavior for the current test, because we create new wallets below for checking balances. However, this change is important with dynamic fees because the transactions that are issued later in the file will produce a change output. The default change output is selected from any of the provided keys (which can break the balance checks).

This isn't a problem in the current test because the static fee for all the Primary Network staking is `0` - which means we never include a change output. This is because the wallet consumes locked UTXOs first for staking operations, and the remaining tokens from those transactions must use the same owner that it was consumed from.